### PR TITLE
xlibContext

### PR DESCRIPTION
**Pull request by fundies**

### CODE REVIEW

**Comment by RobertBColton** [comment]
What's the point of this pull?

**Comment by fundies** [comment]
Creates context with proper attributes

File: ENIGMAsystem/SHELL/Bridges/xlib-OpenGL1/attribs.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include <GL/gl.h>
+#include <GL/glx.h>
+
+namespace enigma {
+
+int context_attribs[] = {
+    GLX_CONTEXT_MAJOR_VERSION_ARB, 1,
+    GLX_CONTEXT_MINOR_VERSION_ARB, 1,
+    GLX_CONTEXT_PROFILE_MASK_ARB, GLX_CONTEXT_COMPATIBILITY_PROFILE_BIT_ARB,
+    None
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Bridges/xlib-OpenGL1/attribs.cpp:10]: This should break everything, even though you are requesting a compatibility profile, this makes it always be version 1.1 so surfaces will never work. Try a game with surfaces to confirm.

File: ENIGMAsystem/SHELL/Bridges/xlib-OpenGL3/attribs.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace enigma {
+
+int context_attribs[] = {
+    GLX_CONTEXT_MAJOR_VERSION_ARB, 3,
+    GLX_CONTEXT_MINOR_VERSION_ARB, 3,
+    GLX_CONTEXT_PROFILE_MASK_ARB, GLX_CONTEXT_CORE_PROFILE_BIT_ARB,
+    #ifdef DEBUG_MODE
+    GLX_CONTEXT_FLAGS_ARB, GLX_CONTEXT_DEBUG_BIT_ARB,
+    #endif
```
Comment (RobertBColton) [inline ENIGMAsystem/SHELL/Bridges/xlib-OpenGL3/attribs.cpp:12]: This debug should go in the 1.1 bridge too.

File: ENIGMAsystem/SHELL/Bridges/xlib-OpenGL1/attribs.cpp (hunk 1 above)
Comment (fundies) [reply ENIGMAsystem/SHELL/Bridges/xlib-OpenGL1/attribs.cpp:10]: I dont have any games other than catch clown. And i just guessed at ogl 1 version. What version you want there?

File: ENIGMAsystem/SHELL/Bridges/xlib-OpenGL3/attribs.cpp (hunk 2 above)
Comment (fundies) [reply ENIGMAsystem/SHELL/Bridges/xlib-OpenGL3/attribs.cpp:12]: there is no debug context till ogl 3

**Pull request merged**